### PR TITLE
Remove unnecessary dummy span checks

### PR DESCRIPTION
**Pull request by estebank**
The emitter already verifies wether a given span note or span label
can be emitted to the output. If it can't, because it is a dummy
span, it will be either elided for labels or emitted as an unspanned
note/help when applicable.

### CODE REVIEW

**Comment by estebank** [comment]
CC #55669

**Comment by varkor** [comment]
Sorry, I was putting off looking at this because I saw it was marked as "WIP". Is this ready to review now?

**Comment by estebank** [comment]
I'll come back to this soon. I need to add some code to special case the emitter so the `note`s appear indented instead as left aligned when `span == DUMMY_SP`.

**Comment by stokhos** [comment]
Ping from triage @estebank Have you been able to make any progress on this

**Comment by estebank** [comment]
@stokhos completed

@varkor ready for review

File: src/libsyntax_pos/lib.rs [hunk 1]
```diff
@@ -668,6 +668,11 @@ impl MultiSpan {
         &self.primary_spans
     }
 
+    /// Returns wether any of the primary spans is displayable.
+    pub fn has_primary_spans(&self) -> bool {
+        self.primary_spans.iter().any(|sp| *sp != DUMMY_SP)
```
Comment (petrochenkov) [inline src/libsyntax_pos/lib.rs]: Comparison with `DUMMY_SP` is almost never correct, it's preferable to use `span.is_dummy()`.

Comment (estebank) [reply src/libsyntax_pos/lib.rs]: Fixed

File: src/libsyntax_pos/lib.rs [hunk 2]
```diff
@@ -668,6 +668,11 @@ impl MultiSpan {
         &self.primary_spans
     }
 
+    /// Returns wether any of the primary spans is displayable.
```
Comment (varkor) [inline src/libsyntax_pos/lib.rs]:
```suggestion
    /// Returns whether any of the primary spans is displayable.
```

File: src/libsyntax_pos/lib.rs [hunk 3]
```diff
@@ -726,6 +731,11 @@ impl MultiSpan {
 
         span_labels
     }
+
+    /// Returns wether any of the span labels is displayable.
```
Comment (varkor) [inline src/libsyntax_pos/lib.rs]:
```suggestion
    /// Returns whether any of the span labels is displayable.
```

**Review by varkor — APPROVED** [review]
Looks good to me!

**Pull request merged**